remoteworkitem/trackeritem_repository.go

```diff
@@ -33,39 +33,39 @@ func Upload(db *gorm.DB, tID uuid.UUID, item TrackerItemContent) error {
 }
 
 // Map a remote work item into an WIT work item and persist it into the database.
-func ConvertToWorkItemModel(ctx context.Context, db *gorm.DB, tID uuid.UUID, item TrackerItemContent, providerType string, spaceID uuid.UUID) (*workitem.WorkItem, error) {
+func ConvertToWorkItemModel(ctx context.Context, db *gorm.DB, item TrackerItemContent, tq TrackerSchedule) (*workitem.WorkItem, error) {
 	remoteID := item.ID
 	content := string(item.Content)
-	trackerItem := TrackerItem{Item: content, RemoteItemID: remoteID, TrackerID: tID}
+	trackerItem := TrackerItem{Item: content, RemoteItemID: remoteID, TrackerID: tq.TrackerID}
 	// Converting the remote item to a local work item
-	remoteTrackerItemConvertFunc, ok := RemoteWorkItemImplRegistry[providerType]
+	remoteTrackerItemConvertFunc, ok := RemoteWorkItemImplRegistry[tq.TrackerType]
 	if !ok {
-		return nil, BadParameterError{parameter: providerType, value: providerType}
+		return nil, BadParameterError{parameter: tq.TrackerType, value: tq.TrackerType}
 	}
 	remoteTrackerItem, err := remoteTrackerItemConvertFunc(trackerItem)
 	if err != nil {
 		return nil, InternalError{simpleError{message: fmt.Sprintf(" Error parsing the tracker data: %s", err.Error())}}
 	}
-	remoteWorkItem, err := Map(remoteTrackerItem, RemoteWorkItemKeyMaps[providerType])
+	remoteWorkItem, err := Map(remoteTrackerItem, RemoteWorkItemKeyMaps[tq.TrackerType])
 	if err != nil {
 		return nil, ConversionError{simpleError{message: fmt.Sprintf("Error mapping to local work item: %s", err.Error())}}
 	}
-	workItem, err := lookupIdentities(ctx, db, remoteWorkItem, providerType, spaceID)
+	workItem, err := lookupIdentities(ctx, db, remoteWorkItem, tq)
 	if err != nil {
 		return nil, InternalError{simpleError{message: fmt.Sprintf("Error bind assignees: %s", err.Error())}}
 	}
 	return upsert(ctx, db, *workItem)
 }
 
 // lookupIdentities looks up creator and assignee remote identities to local identities (already existing or to be created)
```

Comment:
Please create a better documentation line. It makes no sense to me :)

Reply:
The function did much more than `lookupIdentities`. Changed the function name here - 6648376

Reply:
@DhritiShikhar `// setLocalWorkItemFields sets fields of local workitem` isn't really a good explanation either.

Reply:
@kwk renamed here 
6ddb69d